apps/website/src/components/courses/CourseDirectory.tsx

```diff
@@ -0,0 +1,149 @@
+  { value: 'Intermediate', label: 'Intermediate' },
+  { value: 'Advanced', label: 'Advanced' },
+];
+
+export type CourseDirectoryProps = {
+  displayData: GetCoursesResponse | undefined;
+  displayLoading: boolean;
+  displayError: AxiosError<unknown, unknown> | null | undefined;
+  noResults: boolean;
+  refetch: (filters: CoursesRequestBody) => void;
+};
+
+const CourseDirectory: FC<CourseDirectoryProps> = ({
+  displayData,
+  displayLoading,
+  displayError,
+  noResults,
+  refetch,
+}) => {
+  const [selectedCadences, setSelectedCadences] = useState<string[]>(CADENCE_OPTIONS.map((c) => c.value));
+  const [selectedLevels, setSelectedLevels] = useState<string[]>(LEVEL_OPTIONS.map((l) => l.value));
+  const [filtersOpenMobile, setFiltersOpenMobile] = useState(false);
+
+  const handleCadenceChanged = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value, checked } = event.target;
+    const newSelectedCadences = checked ? [...selectedCadences, value] : selectedCadences.filter((l) => l !== value);
+    setSelectedCadences(newSelectedCadences);
+    refetch({ cadence: newSelectedCadences, level: selectedLevels });
+  };
+
+  const handleLevelChanged = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value, checked } = event.target;
+    const newSelectedLevels = checked ? [...selectedLevels, value] : selectedLevels.filter((l) => l !== value);
+    setSelectedLevels(newSelectedLevels);
+    refetch({ cadence: selectedCadences, level: newSelectedLevels });
+  };
+
+  const showFilterActiveDot = selectedCadences.length !== CADENCE_OPTIONS.length || selectedLevels.length !== LEVEL_OPTIONS.length;
+
+  const renderFilterGroup = useCallback((
+    title: string,
+    options: Option[],
+    selectedValues: string[],
+    handleChange: (event: ChangeEvent<HTMLInputElement>) => void,
+  ) => (
+    <div className="course-directory__filter-group">
+      <P className="font-semibold mb-2">{title}</P>
+      {options.map((option) => (
+        <div key={option.value} className="flex items-center gap-2">
+          <input
+            type="checkbox"
+            id={`${title.toLowerCase()}-${option.value}`}
+            value={option.value}
+            checked={selectedValues.includes(option.value)}
+            onChange={handleChange}
+            className="form-checkbox size-4 text-blue-600 transition duration-150 ease-in-out"
+          />
+          <label htmlFor={`${title.toLowerCase()}-${option.value}`}>
+            {option.label}
+          </label>
+        </div>
+      ))}
+    </div>
+  ), []);
+
+  return (
+    <Section className="course-directory">
+      <div className="course-directory__content flex flex-col md:grid gap-spacing-x grid-cols-[minmax(min-content,15%)_1fr]">
+        <div className="course-directory__filter-section flex flex-col gap-4 w-full">
+          <div className="course-directory__filter-header flex md:hidden justify-between items-center">
+            <H3>Results</H3>
```

Comment:
This only appears on mobile, and is just there to fill some dead space. This is what it would look like without it:
![Screenshot 2025-04-28 at 02 13 06](https://github.com/user-attachments/assets/423c4353-2f87-450a-9280-cedc28b1a47c)